mempool/cache_bench_test.go

```diff
@@ -17,7 +19,7 @@ func BenchmarkCacheInsertTime(b *testing.B) {
 	b.ResetTimer()
 
 	for i := 0; i < b.N; i++ {
-		cache.Push(txs[i])
+		cache.Push(types.Tx(txs[i]).Key())
 	}
```

Comment:
nit: This is a slightly incorrect way of performing the benchmark. The `Key()` method should be done outside of the benchmark the same way that the constructor is made. In fact you just need to make random 32 byte slices (i.e. they don't have to be actual hashes so long as they are unique)